File: MatchUpAPI/src/main/java/com/matchup/api/matchup_api/dtos/UserDTO.java
package com.matchup.api.matchup_api.dtos;

import com.matchup.api.matchup_api.models.Country;
import com.matchup.api.matchup_api.models.Sport;
import com.matchup.api.matchup_api.models.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;
import java.util.stream.Collectors;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserDTO {
    private UUID id;
    private String name;
    private String email;
    private String city;
    private String mobilePhone;
    private String gender;
    private Country country;
    private Sport favoriteSport;
    private List<EnrollmentDTO> enrollments;

    public static UserDTO fromEntity(User u) {
        return new UserDTO(
                u.getId(),
                u.getName(),
                u.getEmail(),
                u.getCity(),
                u.getMobilePhone(),
                u.getGender(),
                u.getCountry() != null ? u.getCountry() : null,
                u.getFavoriteSport() != null ? u.getFavoriteSport() : null,
                u.getEnrollments() != null
                        ? u.getEnrollments().stream()
                        .map(EnrollmentDTO::fromEntity)
                        .collect(Collectors.toList())
                        : null
        );
    }
}

File: MobileApp/app/src/main/java/com/app/matchup/dtos/UserDTO.kt
package com.app.matchup.dtos

import com.app.matchup.models.Country
import com.app.matchup.models.Sport
import java.util.UUID

data class UserDTO (
    val id: UUID,
    val name: String,
    val email: String,
    val passwordHash: String,
    val city: String,
    val mobilePhone: String,
    val gender: String,
    val country: Country,
    val favoriteSport: Sport,
    val enrollments: List<EnrollmentDTO>
)
